### PR TITLE
Container cluster upgrade guide pt 3

### PR DESCRIPTION
**Pull request by c2thorn**
upgrade guide for https://github.com/GoogleCloudPlatform/magic-modules/pull/9067 and
https://github.com/GoogleCloudPlatform/magic-modules/pull/9013

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

File: mmv1/third_party/terraform/website/docs/guides/version_5_upgrade.html.markdown [hunk 1]
```diff
@@ -374,6 +374,42 @@ proposed value to configuration (below) or apply `lifecycle.ignore_changes` to t
 +    }
 ```
 
+### Cluster deletion now prevented by default with `deletion_protection`
+
+The field `deletion_protection` has been added with a default of `true`. This field prevents
+Terraform from destroying the resource or recreating it.
```
Comment (melinath) [inline mmv1/third_party/terraform/website/docs/guides/version_5_upgrade.html.markdown:380]:
```suggestion
Terraform from destroying or recreating the cluster.
```

Comment (melinath) [reply mmv1/third_party/terraform/website/docs/guides/version_5_upgrade.html.markdown:380]: Alternative:

```suggestion
Terraform from destroying the cluster, including due to modification of immutable fields.
```

File: mmv1/third_party/terraform/website/docs/guides/version_5_upgrade.html.markdown [hunk 2]
```diff
@@ -374,6 +374,42 @@ proposed value to configuration (below) or apply `lifecycle.ignore_changes` to t
...
+**`deletion_protection` does NOT prevent deletion outside of Terraform.**
+To destroy a `google_container_cluster`, this field must be explicitly set to `false`.
+
+### `networking_mode` defaults to `VPC_NATIVE` for newly created clusters
```
Comment (melinath) [inline mmv1/third_party/terraform/website/docs/guides/version_5_upgrade.html.markdown:384]: I found this section a bit hard to parse but I'm not sure how to improve it; the change looks complex.

**Review by melinath — APPROVED** [review]

File: mmv1/third_party/terraform/website/docs/guides/version_5_upgrade.html.markdown (hunk 2 above)
Comment (c2thorn) [reply mmv1/third_party/terraform/website/docs/guides/version_5_upgrade.html.markdown:384]: I'll remove this section for now, as it has not been approved yet.
